iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Authentication;
+using Microsoft.Azure.Devices.Generated;
+using Microsoft.Rest;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// /// The Digital Twins Service Client contains methods to retrieve and update digital twin information, and invoke commands on a digital twin device.
+    /// </summary>
+    public class DigitalTwinClient : IDisposable
+    {
+        private readonly IotHubGatewayServiceAPIs _client;
+        private readonly DigitalTwin _protocolLayer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DigitalTwinClient"/> class.</summary>
+        /// <param name="connectionString">Your IoT hub's connection string.</param>
+        public static DigitalTwinClient CreateFromConnectionString(string connectionString)
+        {
+            connectionString.ThrowIfNullOrWhiteSpace(nameof(connectionString));
+
+            var iotHubConnectionString = IotHubConnectionString.Parse(connectionString);
+            var sharedAccessKeyCredential = new SharedAccessKeyCredentials(connectionString);
+            return new DigitalTwinClient(iotHubConnectionString.HttpsEndpoint, sharedAccessKeyCredential);
+        }
+
+        private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials)
+        {
+            _client = new IotHubGatewayServiceAPIs(uri, credentials);
+            _protocolLayer = new DigitalTwin(_client);
+        }
+
+        /// <summary>
+        /// Gets a strongly-typed digital twin asynchronously.
+        /// </summary>
+        /// <param name="digitalTwinId">The Id of the digital twin.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The application/json digital twin and the http response.</returns>
+        public async Task<HttpOperationResponse<T, DigitalTwinGetHeaders>> GetAsync<T>(string digitalTwinId, CancellationToken cancellationToken = default)
```

Comment:
If PnP adds other methods to this client, like for getting models and such, `GetAsync` is going to not be obvious anymore. Any reason not to just use `GetDigitalTwinAsync`?